examples/mm/stable_diffusion/anneal_sdxl.py

```diff
@@ -0,0 +1,323 @@
+
+        def checkpoint_check_fn(module):
+            return isinstance(module, (Decoder, UNetModel, MegatronCLIPRewardModel))
+
+        apply_activation_checkpointing(
+            ptl_model, checkpoint_wrapper_fn=non_reentrant_wrapper, check_fn=checkpoint_check_fn
+        )
+
+    optimizer, scheduler = extract_optimizer_scheduler_from_ptl_model(ptl_model)
+
+    ckpt_callback = add_custom_checkpoint_callback(trainer, ptl_model)
+
+    logger.log_hyperparams(OmegaConf.to_container(cfg))
+
+    torch.distributed.barrier()
+
+    ckpt_callback = add_custom_checkpoint_callback(trainer, ptl_model)
+    timer = Timer(cfg.exp_manager.get("max_time_per_run", None) if cfg.exp_manager else None)
+
+    draft_p_trainer = SupervisedTrainer(
+        cfg=cfg.trainer.draftp_sd,
+        model=ptl_model,
+        optimizer=optimizer,
+        scheduler=scheduler,
+        train_dataloader=val_dataloader,
+        val_dataloader=val_dataloader,
+        test_dataloader=[],
+        logger=logger,
+        ckpt_callback=ckpt_callback,
+        run_timer=timer,
+        run_init_validation=True,
+    )
+
+    if custom_trainer_state_dict is not None:
+        draft_p_trainer.load_state_dict(custom_trainer_state_dict)
+
+    torch.cuda.empty_cache()
+
+    if cfg.get("prompt") is not None:
+        logging.info(f"Override val dataset with custom prompt: {cfg.prompt}")
+        val_dataloader = [[cfg.prompt]]
+
+    wt_types = cfg.get("weight_type", None)
+    if wt_types is None:
+        wt_types = ["base", "draft", "linear", "power_2", "power_4", "step_0.6"]
+    else:
+        wt_types = wt_types.split(",") if isinstance(wt_types, str) else wt_types
+    logging.info(f"Running on types: {wt_types}")
+
+    # run for all weight types
+    for wt_type in wt_types:
+        global_idx = 0
+        if wt_type == "base":
+            # dummy function that assigns a value of 0 all the time
+            logging.info("using the base model")
+            wt_draft = lambda sigma, sigma_next, i, total: 0
+        elif wt_type == "linear":
+            wt_draft = lambda sigma, sigma_next, i, total: i * 1.0 / total
+        elif wt_type == "draft":
+            wt_draft = lambda sigma, sigma_next, i, total: 1
+        elif wt_type.startswith("power"):  # its of the form power_{power}
+            pow = float(wt_type.split("_")[1])
+            wt_draft = lambda sigma, sigma_next, i, total: (i * 1.0 / total) ** pow
+        elif wt_type.startswith("step"):  # use a step function (step_{p})
+            frac = float(wt_type.split("_")[1])
+            wt_draft = lambda sigma, sigma_next, i, total: float((i * 1.0 / total) >= frac)
+        else:
+            raise ValueError(f"invalid weighing type: {wt_type}")
+        logging.info(f"using weighing type for annealed outputs: {wt_type}.")
+
+        # initialize generator
+        exp_dir = cfg.exp_manager.explicit_log_dir
```

Comment:
To avoid repeating yourself:
```suggestion
        annealed_out_dir = os.path.join(cfg.exp_manager.explicit_log_dir, f"annealed_outputs_sdxl_{wt_type})
```

and just reference that variable everywhere